src/commands/active-repeat.js

```diff
@@ -1,26 +1,32 @@
 import State from '../state'
 
 class ActiveRepeat {
-  constructor(repeatService) {
+  constructor({ repeatService, messageService }) {
     this.repeatService = repeatService
+    this.messageService = messageService
     this.state = State.ACTIVE_REPEAT
   }
 
-  shouldExecute(messageService) {
-    if (messageService.getCurrentState() === this.state) {
+  shouldExecute() {
+    const { userEmail } = this.messageService.getMessageData()
```

Comment:
These few lines should just be one function to make it cleaner

Reply:
done in this file, will move to other files once this pr is merged and tested